### PR TITLE
[DRAFT] How to debug the Elastic-Agent

### PR DESCRIPTION
**Pull request by AndersonQ**
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by blakerouse — COMMENTED** [review]
I really like where this is going, being able to attach dlv to a subprocess is really nice. The downside is that it makes the assumption that all are built with golang. We do not actually have a language requirement, how will that be handled? When a subprocess is not using golang?

File: README.md [hunk 1]
```diff
@@ -3,10 +3,32 @@
...
+- [Make](https://www.gnu.org/software/make/)
+
+## Building
+
+To build and package (generate the tar/zip/whatever with the agent, beats and everything else it needs):
+```shell
+DEV=true SNAPSHOT=true EXTERNAL=true PLATFORMS="linux/amd64" PACKAGES="tar.gz" mage -v clean package
+```
+Where:
+ - `PLATFORMS` defines to which platform to build and package
```
[the remark this answers is not in the document]
Comment (AndersonQ) [reply README.md:18]: As far as I know, right now, they code is best place to check it :/

File: README.md [hunk 2]
```diff
@@ -3,10 +3,32 @@
...
+ - `PACKAGES` defines which packages to generate.
+ - `EXTERNAL` if set to true, the beats will be downloaded, they'll be built otherwise.
+ - `SNAPSHOT` if set to true the agent will not verify the beats/applications signatures
```
[the remark this answers is not in the document]
Comment (AndersonQ) [reply README.md:21]: you're right, my bad

File: README.md [hunk 3]
```diff
@@ -3,10 +3,32 @@
...
-- installed [mage](https://github.com/magefile/mage)
+- [Mage](https://github.com/magefile/mage)
+- [Go](https://go.dev/dl/)
+- [Make](https://www.gnu.org/software/make/)
+
+## Building
+
+To build and package (generate the tar/zip/whatever with the agent, beats and everything else it needs):
+```shell
+DEV=true SNAPSHOT=true EXTERNAL=true PLATFORMS="linux/amd64" PACKAGES="tar.gz" mage -v clean package
```
[the remark this answers is not in the document]
Comment (AndersonQ) [reply README.md:15]: true (I always forget that). I'm just thinking that here, it's worth to mention the `DEV` variable to make it known.

**Comment by AndersonQ** [comment]
closing it. The how-to was moved to internal docs, and the changes on the agent are no longer valid

**Pull request closed without merging**